File: packages/fluent-cards/src/builders/TextBlockBuilder.ts
import type { TextBlock } from '../models.js';
import { HorizontalAlignment, TextColor, TextSize, TextWeight } from '../enums.js';

/** Fluent builder for {@link TextBlock} elements. */
export class TextBlockBuilder {
  private readonly block: TextBlock = { type: 'TextBlock', text: '' };

  withId(id: string): this { this.block.id = id; return this; }
  withText(text: string): this { this.block.text = text; return this; }
  withSize(size: TextSize): this { this.block.size = size; return this; }
  withWeight(weight: TextWeight): this { this.block.weight = weight; return this; }
  withColor(color: TextColor): this { this.block.color = color; return this; }
  withWrap(wrap: boolean): this { this.block.wrap = wrap; return this; }
  withMaxLines(maxLines: number): this { this.block.maxLines = maxLines; return this; }
  withHorizontalAlignment(alignment: HorizontalAlignment): this { this.block.horizontalAlignment = alignment; return this; }
  withSpacing(spacing: string): this { this.block.spacing = spacing; return this; }
  withSeparator(separator: boolean): this { this.block.separator = separator; return this; }
  withIsVisible(isVisible: boolean): this { this.block.isVisible = isVisible; return this; }

  build(): TextBlock { return this.block; }
}

File: java/src/main/java/io/fluentcards/TextBlockBuilder.java
package io.fluentcards;

import java.util.*;

public class TextBlockBuilder {
    private final LinkedHashMap<String, Object> data;

    public TextBlockBuilder() {
        data = new LinkedHashMap<>();
        data.put("type", "TextBlock");
        data.put("text", "");
    }

    public TextBlockBuilder withId(String id) {
        data.put("id", id);
        return this;
    }

    public TextBlockBuilder withText(String text) {
        data.put("text", text);
        return this;
    }

    public TextBlockBuilder withSize(TextSize size) {
        data.put("size", size.getValue());
        return this;
    }

    public TextBlockBuilder withWeight(TextWeight weight) {
        data.put("weight", weight.getValue());
        return this;
    }

    public TextBlockBuilder withColor(TextColor color) {
        data.put("color", color.getValue());
        return this;
    }

    public TextBlockBuilder withIsSubtle(boolean isSubtle) {
        data.put("isSubtle", isSubtle);
        return this;
    }

    public TextBlockBuilder withSubtle() {
        data.put("isSubtle", true);
        return this;
    }

    public TextBlockBuilder withWrap(boolean wrap) {
        data.put("wrap", wrap);
        return this;
    }

    public TextBlockBuilder withMaxLines(int maxLines) {
        data.put("maxLines", maxLines);
        return this;
    }

    public TextBlockBuilder withHorizontalAlignment(HorizontalAlignment alignment) {
        data.put("horizontalAlignment", alignment.getValue());
        return this;
    }

    public TextBlockBuilder withFontType(FontType fontType) {
        data.put("fontType", fontType.getValue());
        return this;
    }

    public TextBlockBuilder withStyle(TextBlockStyle style) {
        data.put("style", style.getValue());
        return this;
    }

    public TextBlockBuilder withSpacing(Spacing spacing) {
        data.put("spacing", spacing.getValue());
        return this;
    }

    public TextBlockBuilder withSeparator(boolean separator) {
        data.put("separator", separator);
        return this;
    }

    public TextBlockBuilder withIsVisible(boolean isVisible) {
        data.put("isVisible", isVisible);
        return this;
    }

    public TextBlockBuilder withSelectAction(Map<String, Object> action) {
        data.put("selectAction", action);
        return this;
    }

    public Map<String, Object> build() {
        return data;
    }
}
